Add minutes from Sept 13, 2021 authentication panel meeting (#193)
solid/authentication-panel: meetings/2021-09-13.md

https://meet.jit.si/solid-authentication

## Agenda

* Feedback from CSS developers
* [PR #43](https://github.com/solid/solid-oidc/pull/43) Rename Client WebID to Client Identifier - any final thoughts?
* client credentials flow
  * [RFC7521: 4.2. Using Assertions for Client Authentication](https://www.rfc-editor.org/rfc/rfc7521.html#section-4.2)

## Present

* Aaron Coburn
* Joachim Van Herwegen
* elf Pavlik
* Nicolas Seydoux
* Barath Raghavan

## Minutes

### Feedback from CSS developers

Pavlik: I understand that CSS implements both the Resource Server (RS) and the Identity Provider (IdP) portions of Solid-OIDC

Joachim: That is correct: there is an IdP component, as well as access control enforcement on the RS side.

Pavlik: From the IdP side, do you implement everything from the current specification?

Joachim: I have implemented Client Identifiers but not the public one (`solid:PublicOidcClient`), because it is a bit of a hassle when working with the existing OIDC-library.

Pavlik: There is an open issue to reconsider if this Public Client Identifier should be kept in the spec. The fact that it is a hassle to implement is an interesting bit of feedback.

> Aaron: The purpose of the Public Client Identifier is to be a stepping stone from the current use of ephemeral clients (dynamic client registration) to an ecosystem where clients are unambiguously identified (i.e. via Client Identifiers).

Reviewer: ```suggestion
Aaron: The purpose of the Public Client Identifier is to be a stepping stone from the current use of ephemeral clients (dynamic client registration) to an ecosystem where clients are unambiguously identified (e.g., via Client Identifiers).
```

Joachim: The issue is that all clients end up with the same Client ID, which is problematic in some cases.

Aaron: The problems (motivating its removal) are the following:
* The URI isn't dereferencable
* For an anonymous client, what is the value proposition of using the Public Client Identifier? Using the same identifier for all clients means you cannot disambiguate between apps.

> Joachim: The concept of having anonymous clients having less privileges is a good idea though.

Reviewer: ```suggestion
Joachim: Making anonymous clients have less privileges is a good idea though.
```

Aaron: One use case is worth considering: how can you distinguish between an ephemeral client and regular client credentials?

> If such a distinction is necessary in some cases, then it could make sense to use an well-known client identiier.

Reviewer: ```suggestion
If such a distinction is necessary in some cases, then it could make sense to use a well-known client identifier.
```

_Aaron describes the client credential flow to make sure we all talk about the same thing_

Nicolas: OIDC spec recommends that ID token which you get back gets client id in `sub` claim for `client_credentials` flow
*edit*: It is actually a contraint on the access token: https://datatracker.ietf.org/doc/html/draft-ietf-oauth-access-token-jwt-12#section-2.2 and part of a separate (draft) specification

> Joachim: In the diagram flows, what is the relationship between a resource server (RS) and an authorization server (AS)

Reviewer: ```suggestion
Joachim: In the diagram flows, what is the relationship between a resource server (RS) and an authorization server (AS)?
```
